File: loops/loops_notes.py
# Adelle Ponczoch, Loop Notes in Python

# Adelle Ponzoch, Loops Notes in Python

# What is a loop? 
    # a section of code that is going to repeat
# What are the two types of loops?
    #while loop
x = 0
while x < 10:
    print("Hello", x)
    x +=1

    #for loop
nums = [3,5,7,2,8]
for num in nums:
    print(num)
# What is iteration
    #when you do something multiple times, the act of repeating (version)

# What are lists? 
    #a bunch of values in the same variable
siblings = ["Alex", "Katie", "Andrew", "Vienna", "Tori", "Treyson", "Jeff", "Hailey"]
#print one item from the list
print(siblings[3])
#change an item in a list
siblings[7] = "Jake"

#remove item in list
siblings.pop(5)
print(siblings)
#proper list printing with a for loop
num = 1
for sibling in siblings:
    print(f"{num}. {sibling} LaRose")
    num +=1

# How do you make lists in python? 
    #you make a variable, do brackets and put commas in between list items

# How do you make for loops in python? 
for num in range(1,11, 2):
    print(num)
# How do you make while loops in python?
num = 1

while num <21:
    print(num)
    num += 1

import random

num = 1
rand = random.randint(1,20)
while num < 21:
    if num == rand:
        print(f"Goose!")
        break # Tells the loop to end once the conditon is met
    else:
        print("Duck")
    num += 1

# Continue tells the loop to stop that particular round of the loop

File: loops/loops_notes.c
//Adelle Ponczoch, Loops Notes in C
#include <stdio.h>

int main(void){
    //What is a loop? 
        // a section of code that repeats
    //What are the two types of loops?
        //for loops
        int x;
        for(x=0; x<10; x++){
            printf("Hello");
        }
        //while loops
        int i=1;
        while(i<10){
            printf("%d\n", i);
            i++;
        }
    //What is iteration
        //the act of repeating (a specific time through the loop])
    //What are lists? (called arrays in c)
        //many values in the same variable
    //How do you make lists in C?
        int grades[] = {97, 95, 100, 70, 94, 98, 64};
        // in the brackets we say how long the list is going to be, but don't need it if you set the list right there
        // data type is whatever is in the list, all have to be the same data type
        // list is surrounded by curly brackets with commas between each items
        printf("%d\n", grades[3]); //we put the index number in the brackets when we print
        grades[2] = 73; //update one at a time
        printf("%d\n", grades[2]);
        // How to get the size of an array (list)
        //printf("%lu", sizeof(grades));//gives number of bytes that the list takes
        int length = sizeof(grades)/sizeof(grades[0]);
        printf("%d\n", length);
    //How do you make for loops in C? //iterators should be x or i
        int t; 
        //in parentheses 1. starting point 2. go until point 3. what we count by
        for(t=0; t<10; t+=2){
            printf("%d\n", t);
        }
        int l;
        for(l=0; l<length; l++){
            printf("%d\n", grades[l]);
        }
    //How do you make while loops in C?
    // use iterator to se start point
    // while line sets the stop point
    // last line set what you count by
        int iterator = 100;
        while(iterator >= 0){
            printf("%d\n", iterator);
            iterator -= 10;
        }
        char movies[][20] = {"Cinderella", "The Smurf Movie", "Transformers", "Cars", "Up", "1984"};
        int mlength = sizeof(movies)/sizeof(movies[0]);
        int m = 0;
        while(m<mlength){
            printf("%s\n", movies[m]);
            m++;
        }

    return 0;
}